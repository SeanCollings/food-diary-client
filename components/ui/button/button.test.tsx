import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './';

describe('<Button/>', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Button height={50} width={150}>
        Test
      </Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render in loading state', () => {
    const { asFragment } = render(<Button loading>Test</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const { asFragment } = render(
      <Button background="secondary" borderWidth={8} fontSize={12} color="red">
        Test
      </Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onclick', async () => {
    const mockOnClick = jest.fn();

    render(<Button onClick={mockOnClick}>Test</Button>);

    fireEvent.click(await screen.findByRole('button', { name: 'Test' }));

    expect(mockOnClick).toHaveBeenCalled();
  });
});
