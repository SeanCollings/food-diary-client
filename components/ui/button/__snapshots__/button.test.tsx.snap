// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render 1`] = `
<DocumentFragment>
  <div
    class="sc-c8087764-0 eiHINF"
    width="150"
  >
    <button
      class="sc-c8087764-1 hpCTkN"
      color="var(--text)"
      font-size="24"
      height="50"
      width="150"
    >
      <div
        class="sc-c8087764-2 hwgIAl"
      >
        <div
          class="sc-c8087764-4 btSOSc"
        >
          Test
        </div>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Button/> should render in loading state 1`] = `
<DocumentFragment>
  <div
    class="sc-c8087764-0 eHJkMl loading"
  >
    <button
      class="sc-c8087764-1 eYFVmU loading"
      color="var(--text)"
      disabled=""
      font-size="24"
    >
      <div
        class="sc-c8087764-2 hwgIAl"
      >
        <div
          class="sc-c8087764-3 gpmtlB loading"
        >
          <div
            class="sc-4731ba6b-0 cnZVOU loader"
            color="var(--text)"
          />
        </div>
        <div
          class="sc-c8087764-4 btSOSc loading"
        >
          Test
        </div>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Button/> should render with custom props 1`] = `
<DocumentFragment>
  <div
    class="sc-c8087764-0 jNTqvr"
  >
    <button
      class="sc-c8087764-1 loRlQM"
      color="red"
      font-size="12"
    >
      <div
        class="sc-c8087764-2 hwgIAl"
      >
        <div
          class="sc-c8087764-4 btSOSc"
        >
          Test
        </div>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
